src/components/RuleItem.vue

```diff
@@ -430,6 +431,55 @@ async function updateRuleShipping(event: any) {
     showToast(translate("Failed to update rule brokering."))
   }
 }
+
+function isReorderPossible(ruleDir: string, ruleIndex: any) {
+  if(ruleDir === 'prev' && ruleIndex === 0) return false;
+  if(ruleDir === 'next' && ruleIndex === rules.value.length - 1) return false;
+
+  return true;
+}
+
+async function updateRuleOrder(ruleDir: string) {
+  const prevSeq = JSON.parse(JSON.stringify(rules.value));
+  const updatedSeq = JSON.parse(JSON.stringify(rules.value));
+  let alternateRuleIndex = '' as any;
+
+  if(ruleDir === 'prev') alternateRuleIndex = props.ruleIndex - 1
+  else alternateRuleIndex = props.ruleIndex + 1;
+
+  [updatedSeq[props.ruleIndex], updatedSeq[alternateRuleIndex]] = [updatedSeq[alternateRuleIndex], updatedSeq[props.ruleIndex]]
+
+  let diffSeq = findRulesDiff(prevSeq, updatedSeq)
+
+  const updatedSeqenceNum = prevSeq.map((rule: any) => rule.sequenceNum)
+  Object.keys(diffSeq).map((key: any) => {
+    diffSeq[key].sequenceNum = updatedSeqenceNum[key]
+  })
+
+  diffSeq = Object.keys(diffSeq).map((key) => diffSeq[key])
+
+  try {
+    diffSeq.map(async (rule: any) => {
+      await RuleService.updateRule(rule, rule.ruleId);
+    })
+    await store.dispatch('rule/updateRules', { rules: updatedSeq })
+    showToast("Rules order has been updated successfully.")
+  } catch(err: any) {
+    logger.error(err);
+    showToast(translate("Failed to update rules order."))
+  }
+}
+
+function findRulesDiff(previousSeq: any, updatedSeq: any) {
+  const diffSeq: any = Object.keys(previousSeq).reduce((diff, key) => {
+    if (updatedSeq[key].routingRuleId === previousSeq[key].routingRuleId && updatedSeq[key].statusId === previousSeq[key].statusId && updatedSeq[key].assignmentEnumId === previousSeq[key].assignmentEnumId && updatedSeq[key].ruleName === previousSeq[key].ruleName) return diff
```

Comment:
I think there is no property like `routingRuleId` on rules, instead it is `ruleId`.

Also, there is no need to check for `statusId`, `assignmentEnumId` and `ruleName`, as on reorder we are only updating the seqNum.